### ISSUE TITLE
Support running via `-m`

### ISSUE THREAD

**Issue by brettcannon** [pypa/pipenv#1474]
It would be great if `python3 -m pipenv` worked in order to support Windows users better as neither `pipsi` nor `--user` installs are accessible by default. As it currently stands there's a message about `pew` needing to be on the `PATH`, although there's an [environment variable](https://docs.pipenv.org/advanced/#configuration-with-environment-variables) called `PIPENV_VENV_IN_PROJECT` which doesn't seem to have a corresponding command-line flag (I also didn't think to test if this removed the `pew` warning before leaving work today).

**Comment by techalchemy** [comment]
Please see [this comment](https://github.com/pypa/pipenv/issues/913#issuecomment-366456086) -- you need to call pip via `python3 -m` in the first place because you haven't configured your `%PATH%` environment variable, as the error message indicates.

**Comment by uranusjr** [comment]
I think making things work with `-m` is good architecture in general though, whether or not there are alternatives. This would also work around the current workaround we have with the pipsi installation that `pipsi install pew` is required. `[sys.executable, ‘-m’, ‘pew’]` would always work as long as pew is installed during Pipenv’s installation (it is with the current setup).

This would require pew to work that way first.

**Comment by techalchemy** [comment]
@uranusjr I am far from an expert with regard to this aspect of our implementation so I will reopen for now

Do you have a clear understanding of what is preventing pew from working this way?

**Comment by uranusjr** [comment]
It needs to add a `if __name__ == '__main__':` block somewhere, and call the program’s entry point in it. [This is how Pipenv does it](https://github.com/pypa/pipenv/blob/master/pipenv/__main__.py).

pew’s entry point is `pew.pew:pew`, judging from its [configuration](https://github.com/berdario/pew/blob/1.1.2/setup.py#L51) in `setup.py`, so one possible implementation would be to add a `__main__.py` with

```python
from .pew import pew
if __name__ == '__main__':
    pew()
```

**Comment by techalchemy** [comment]
@uranusjr ah I thought you were saying there was something more significant

You should submit a PR upstream!

**Comment by uranusjr** [comment]
I have already too many things at hand. This is literally a three-liner but I just don’t want to go through the human part off submitting patches. Somebody please do that please? Pretty please? <insert cat meme>

**Comment by jtratner** [comment]
I can do it.

**Comment by brettcannon** [comment]
And if @jtratner can't find the time I can try as well.

**Comment by kennethreitz** [comment]
Requiring pew to be installed with the same version of python would be a pain — let's fall back to a system-available pew if `-m pew` doesn't work.

**Comment by kennethreitz** [comment]
not necessary, just being thorough.

**Comment by jtratner** [comment]
PR up - https://github.com/berdario/pew/pull/177

**Comment by kennethreitz** [comment]
merged :)

**Comment by kennethreitz** [comment]
waiting on a release

**Comment by techalchemy** [comment]
With the current release:

```cmd
C:\Users\Dan.Ryan>pipenv --version
pipenv, version 11.0.2

C:\Users\Dan.Ryan>py -3 -m pipenv --version
pipenv, version 11.0.2
```

/cc @brettcannon 

Thanks for the contributions all :fireworks:

**Comment by uranusjr** [comment]
@techalchemy Most operations wouldn’t work though, unless you also expose the `pew` executable. That part requires a new Pew release to work.

**Comment by kennethreitz** [comment]
this is not going to happen.

**Comment by kennethreitz** [comment]
i experimented heavily with getting us to run pew directly, and it's just not feasible.

**Comment by kennethreitz** [comment]
so, we're going to have to stick with subprocessing it out and expecting it to be in the PATH.

**Comment by kennethreitz** [comment]
although, it's very fun to write `pew.pew.pew()`

**Comment by uranusjr** [comment]
I don’t mean running it directly, but with subprocess with `python -m pew` instead. This should be doable after berdario/pew#177 is released.

**Comment by kennethreitz** [comment]
ah, i hadn't considered that!

**Comment by brettcannon** [comment]
So should this issue be re-opened to track calling `pew` with `-m`? Or should I open another issue?

**Comment by uranusjr** [comment]
Let’s just track it here.

**Comment by kennethreitz** [comment]
:+1:

**Comment by kennethreitz** [comment]
making this work anyway.

**Comment by kennethreitz** [comment]
done

**Comment by kennethreitz** [comment]
released

**Comment by techalchemy** [comment]
No worries @berdario we included pew in our vendored packages within a few hours of my PR probably mostly to support the `pew.pew.pew()` invocation
